src/sources/datadog/agent.rs

```diff
@@ -73,9 +64,71 @@ impl SourceConfig for DatadogAgentConfig {
     }
 }
 
-#[derive(Clone, Default)]
+#[derive(Clone)]
 struct DatadogAgentSource {
     store_api_key: bool,
+    api_key_matcher: Regex,
+    log_schema_timestamp_key: &'static str,
+    log_schema_source_type_key: &'static str,
+}
+
+impl DatadogAgentSource {
+    fn new(store_api_key: bool) -> Self {
+        Self {
+            store_api_key,
+            api_key_matcher: Regex::new(r"^/v1/input/(?P<api_key>[[:alnum:]]{32})/??").unwrap(),
```

Comment:
```suggestion
            api_key_matcher: Regex::new(r"^/v1/input/(?P<api_key>[[:alnum:]]{32})/??").expect("static regex always compiles"),
```

I realize we were using `unwrap()` prior; I just like that you can provide a reason with `expect`.